data/src/main/java/org/oppia/data/backends/gae/OppiaGaeClient.kt

```diff
@@ -0,0 +1,20 @@
+package org.oppia.data.backends.gae
+
+import okhttp3.OkHttpClient
+
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+
+/** An object that contains the Retrofit configuration with the Oppia backend. */
+object OppiaGaeClient {
```

Comment:
Let's go ahead and move this to the Dagger graph so that we can inject TopicService directly. We probably don't need OppiaGaeClient at all, and we can also hide the retrofit instance. A pseudocode implementation of this is:

```kotlin
@Module
class NetworkModule {
  @Qualifier private annotation class OppiaRetrofit

  @OppiaRetrofit
  @Provides
  @Singleton
  fun provideRetrofitInstance() {
    return retrofit2.Retrofit.Builder().
      baseUrl(...)
      ...
      .build()
  }

  @Provides
  @Singleton
  fun provideTopicService(@OppiaRetrofit retrofit: Retrofit) {
    return retrofit.create(TopicService::class.java)
  }
}
```

NB: This module needs to be included in the application component's module list.

And with this module, we can now inject ``TopicService`` wherever we need it.